cranelift-codegen/meta/src/shared/legalize.rs

```diff
@@ -272,6 +289,59 @@ pub(crate) fn define(insts: &InstructionGroup, imm: &Immediates) -> TransformGro
         ],
     );
 
+    narrow.custom_legalize(icmp_imm, "narrow_icmp_imm");
+
+    let intcc_eq = Literal::enumerator_for(&imm.intcc, "eq");
+    narrow.legalize(
+        def!(b = icmp.I64(intcc_eq, x, y)),
+        vec![
+            def!((xl, xh) = isplit(x)),
+            def!((yl, yh) = isplit(y)),
+            def!(b1 = icmp(intcc_eq, xl, yl)),
+            def!(b2 = icmp(intcc_eq, xh, yh)),
+            def!(b = band(b1, b2)),
+        ],
+    );
+
+    let intcc_ne = Literal::enumerator_for(&imm.intcc, "ne");
+    narrow.legalize(
+        def!(b = icmp.I64(intcc_ne, x, y)),
+        vec![
+            def!((xl, xh) = isplit(x)),
+            def!((yl, yh) = isplit(y)),
+            def!(b1 = icmp(intcc_ne, xl, yl)),
+            def!(b2 = icmp(intcc_ne, xh, yh)),
+            def!(b = bor(b1, b2)),
+        ],
+    );
+
+    for cc in &["sgt", "sge", "slt", "sle", "ugt", "uge", "ult", "ule"] {
+        let intcc_cc = Literal::enumerator_for(&imm.intcc, cc);
+        let cc1 =
+            Literal::enumerator_for(&imm.intcc, Box::leak(without_equal(cc).into_boxed_str()));
+        let cc2 = Literal::enumerator_for(
+            &imm.intcc,
+            Box::leak(inverse_without_equal(cc).into_boxed_str()),
+        );
+        let cc3 = Literal::enumerator_for(&imm.intcc, Box::leak(unsigned(cc).into_boxed_str()));
+        narrow.legalize(
+            def!(b = icmp.I64(intcc_cc, x, y)),
+            vec![
+                def!((xl, xh) = isplit(x)),
+                def!((yl, yh) = isplit(y)),
+                // X = A || (!A && !B && C)
```

Comment:
What's the A / B / C this comment refers to?

Reply:
The results of cc1, cc2 and cc3. This is just a helpful comment to help visualize the final boolean expression.

Depending on what you prefer, I could either remove or expand this comment.